libc-bottom-half/sources/chdir.c

```diff
@@ -8,10 +8,8 @@
 #include <wasi/libc-find-relpath.h>
 #include <wasi/libc.h>
 
-#ifdef _REENTRANT
-#error "chdir doesn't yet support multiple threads"
-#endif
-
+void __wasilibc_cwd_lock(void);
```

Comment:
Wrap these in `#if _REENTRANT`?

We could do `#define __wasilibc_cwd_lock()` in the else block if you want to avoid more ifdefs in the code?